### PR TITLE
feat: PayPal frontend changes

### PR DESCRIPTION
**Pull request by Riddhiagrawal001**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [X] I ran `npm run re:build`
- [X] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+        }
+      }
+    }
+
+    React.useEffect0(() => {
+      initializePayPalWindow()
+      None
+    })
+
+    <button
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res:48]: use the <Button Component

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+      className="!w-fit rounded-md bg-blue-700 text-white py-2 h-fit border px-6 flex items-center justify-center gap-2"
+      onClick={e => {
+        e->ReactEvent.Mouse.stopPropagation
+      }}>
+      <AddDataAttributes attributes=[("data-paypal-button", "true")]>
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res:53]: why we need the AddDataAttribute?

**Review by JeevaRamu0104 — COMMENTED** [review]
Please check the functionality again

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+            onClick={_ => getStatus()->ignore}>
+            {"Refresh Status"->React.string}
+          </p>
+        </div>
+      </UIUtils.RenderIf>
+    </div>
+  }
+}
+
+external toJson: 'a => Js.Json.t = "%identity"
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res:241]: Use Identity file for that

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res (hunk 2 above)
Comment (Riddhiagrawal001) [reply src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res:53]: This attribute is required in the button to open paypal onboarding flow

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res (hunk 1 above)
Comment (Riddhiagrawal001) [reply src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res:48]: we have to use core button component as we have to pass children and Button component doesn't allow

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res (hunk 3 above)
Comment (Riddhiagrawal001) [reply src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res]: It was a unused code , hence removed

File: src/screens/HyperSwitch/Connectors/ConnectorPreview.res [hunk 4]
```diff
@@ -263,6 +267,43 @@ let make = (
...
 
+  let handleStateToNextPage = () => {
+    setCurrentStep(_ => ConnectorTypes.PaymentMethods)
+  }
+  let getStatus = async () => {
+    open PayPalFlowUtils
+    open LogicUtils
+    try {
+      setScreenState(_ => PageLoaderWrapper.Loading)
+      let paypalBody = PayPalFlowUtils.generatePayPalBody(
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Connectors/ConnectorPreview.res:278]: already open

File: src/screens/HyperSwitch/Connectors/ConnectorPreview.res [hunk 5]
```diff
@@ -263,6 +267,43 @@ let make = (
...
+        ~profileId=Some(connectorInfo.profile_id),
+        (),
+      )
+      let url = `${getURL(~entityName=PAYPAL_ONBOARDING, ~methodType=Post, ())}/sync`
+      let responseValue = await updateDetails(url, paypalBody, Fetch.Post)
+      let paypalDict = responseValue->getDictFromJsonObject->getJsonObjectFromDict("paypal")
+      switch paypalDict->Js.Json.classify {
+      | JSONString(str) => {
+          setCurrentStep(_ => IntegFields)
+          setSetupAccountStatus(._ => str->PayPalFlowUtils.stringToVariantMapper)
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Connectors/ConnectorPreview.res:289]: already open

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+        dictOfInitialValues->Dict.set("status", "active"->Js.Json.string)
+        setInitialValues(_ => dictOfInitialValues->Js.Json.object_)
+        handleConnectorConnected(dictOfInitialValues->Js.Json.object_)->ignore
+      }
+    | _ => ()
+    }
+    Js.Nullable.null->Js.Promise.resolve
+  }
+
+  let proceedButton = switch setupAccountStatus {
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res:461]: `<UIUtils.RenderIf condition={setupAccountStatus !==Redirecting_to_paypal}>
 <FormRenderer.SubmitButton
      loadingText="Processing..."
      text="Proceed"
      disabledParamter={configuartionType === NotSelected ? true : false}
    />

</UIUtils.RenderIf>

`


Directly inside Form can be written

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+      loadingText="Processing..."
+      text="Proceed"
+      disabledParamter={configuartionType === NotSelected ? true : false}
+    />
+  }
+
+  <div className="w-full h-full flex flex-col justify-between">
+    <PageLoaderWrapper screenState>
+      <Form initialValues validate={validateMandatoryFieldForPaypal} onSubmit={handleOnSubmit}>
+        <div className="">
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res:474]: empty can be removed

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res [hunk 8]
```diff
@@ -0,0 +1,534 @@
...
+    Js.Nullable.null->Js.Promise.resolve
+  }
+
+  let proceedButton = switch setupAccountStatus {
+  | Redirecting_to_paypal => React.null
+  | _ =>
+    <FormRenderer.SubmitButton
+      loadingText="Processing..."
+      text="Proceed"
+      disabledParamter={configuartionType === NotSelected ? true : false}
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res:467]: not req true false

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res [hunk 9]
```diff
@@ -0,0 +1,534 @@
...
+    HyperswitchAtom.paypalAccountStatusAtom,
+  )
+  let (suggestedAction, suggestedActionExists) = ConnectorUtils.getSuggestedAction(
+    ~verifyErrorMessage,
+    ~connector,
+  )
+
+  let onSubmitMain = async values => {
+    open ConnectorUtils
+    open PayPalFlowUtils
```
Comment (PritishBudhiraja) [inline src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res:296]: It;s open and called multiple places just check once

File: src/screens/HyperSwitch/Connectors/ConnectorPreview.res (hunk 4 above)
Comment (Riddhiagrawal001) [reply src/screens/HyperSwitch/Connectors/ConnectorPreview.res:278]: Resolved

File: src/screens/HyperSwitch/Connectors/ConnectorPreview.res (hunk 5 above)
Comment (Riddhiagrawal001) [reply src/screens/HyperSwitch/Connectors/ConnectorPreview.res:289]: Resolved

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res (hunk 6 above)
Comment (Riddhiagrawal001) [reply src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res]: Wrapped the component with UIUtils than switch case

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res (hunk 7 above)
Comment (Riddhiagrawal001) [reply src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res]: Resolved

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res (hunk 8 above)
Comment (Riddhiagrawal001) [reply src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res]: Resolved

File: src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res (hunk 9 above)
Comment (Riddhiagrawal001) [reply src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res:296]: Resolved

**Review by PritishBudhiraja — APPROVED** [review]

**Review by PritishBudhiraja — DISMISSED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request closed without merging**